From 4908238597c4e6200289d91f15e15cb562405613 Mon Sep 17 00:00:00 2001
From: Marcin Wojtas <mw@semihalf.com>
Date: Tue, 8 Nov 2016 16:37:47 +0100
Subject: [PATCH 1088/2241] arm64: dts: cp110: add unique identifiers for
 Armada 70x0/80x0 SPI controllers

Enabling SPI controllers, which are attached to different busses
inside an SoC, may result in overlapping enumeration and cause
sysfs registration failure. Example log after enabling two
controllers on Armada 8040 SoC, which belong to AP806 and
CP110-slave HW blocks:

[    3.740415] sysfs: cannot create duplicate filename
'/class/spi_master/spi0'
[    3.747510] ------------[ cut here ]------------
[    3.752145] WARNING: at fs/sysfs/dir.c:31
[...]
[    4.002299] orion_spi: probe of f4700600.spi failed with error -17

spi-orion driver offers dedicated DT property ('cell-index'), that
allow setting unique identifiers. This commit fixes the issue
by assigning different 'cell-index' values for all Armada 70x0/80x0
SPI controllers.

Change-Id: Ib71de286bdf53cec93efecf750adabda3b7f8f8e
Signed-off-by: Marcin Wojtas <mw@semihalf.com>
Reviewed-on: http://vgitil04.il.marvell.com:8080/33636
Tested-by: Star_Automation <star@marvell.com>
Reviewed-by: Omri Itach <omrii@marvell.com>
Reviewed-by: Hanna Hawa <hannah@marvell.com>
---
 arch/arm64/boot/dts/marvell/armada-cp110-1.dtsi | 4 ++--
 arch/arm64/boot/dts/marvell/armada-cp110.dtsi   | 4 ++--
 2 files changed, 4 insertions(+), 4 deletions(-)

diff --git a/arch/arm64/boot/dts/marvell/armada-cp110-1.dtsi b/arch/arm64/boot/dts/marvell/armada-cp110-1.dtsi
index 68f208b..01a0db6 100644
--- a/arch/arm64/boot/dts/marvell/armada-cp110-1.dtsi
+++ b/arch/arm64/boot/dts/marvell/armada-cp110-1.dtsi
@@ -81,7 +81,7 @@ spi@700600 {
 	reg = <0x700600 0x50>;
 	#address-cells = <0x1>;
 	#size-cells = <0x0>;
-	cell-index = <0x0>;
+	cell-index = <0x3>;
 	clocks = <&cps_syscon0 1 21>;
 	status = "disabled";
 };
@@ -91,7 +91,7 @@ spi@700680 {
 	reg = <0x700680 0x50>;
 	#address-cells = <0x1>;
 	#size-cells = <0x0>;
-	cell-index = <0x0>;
+	cell-index = <0x4>;
 	clocks = <&cps_syscon0 1 21>;
 	status = "disabled";
 };
diff --git a/arch/arm64/boot/dts/marvell/armada-cp110.dtsi b/arch/arm64/boot/dts/marvell/armada-cp110.dtsi
index aab16b7..91fc217 100644
--- a/arch/arm64/boot/dts/marvell/armada-cp110.dtsi
+++ b/arch/arm64/boot/dts/marvell/armada-cp110.dtsi
@@ -172,7 +172,7 @@ spi@700600 {
 	reg = <0x700600 0x50>;
 	#address-cells = <0x1>;
 	#size-cells = <0x0>;
-	cell-index = <0x0>;
+	cell-index = <0x1>;
 	clocks = <&cpm_syscon0 1 21>;
 	status = "disabled";
 };
@@ -182,7 +182,7 @@ spi@700680 {
 	reg = <0x700680 0x50>;
 	#address-cells = <0x1>;
 	#size-cells = <0x0>;
-	cell-index = <0x0>;
+	cell-index = <0x2>;
 	clocks = <&cpm_syscon0 1 21>;
 	status = "disabled";
 };
-- 
2.7.4

